src/components/NcInputField/NcInputField.vue

```diff
@@ -260,7 +260,7 @@ export default {
 		},
 
 		hasLeadingIcon() {
-			return this.$slots.default
+			return isSlotPopulated(this.$slots.default?.())
```

Comment:
**question (non-blocking):** checking slots

Do we need to change behavior here and check the content of the slot, not only if the slot is passed or not?

Also, it is not reactive, it won't work as a `computed`.

Reply:
I think we nowhere check the slot content. So I would tend to keep it this way.

The reactivity is an issue though. But I think this is a general problem of using `$slots` outside the render function or template in vue 3.
One solution might be to use `isSlotPopulated(this.$slots.default?.())` directly in the template. But it's used three times, which makes it a bit ugly. The other option would be to move to a render function completely, but that would be overkill, I think. Other ideas are welcome, though.

Reply:
What about just keeping slot existence?
```js
return !!this.$slots.default
```

Reply:
That doesn't work in vue 3 anymore (or at least I didn't get it to work). It also exists even if it just contains a comment or a Fragment that won't actually render anything. See here for example: https://github.com/vuejs/core/issues/4733

Reply:
@ShGKme See #4516.

Reply:
> That doesn't work in vue 3 anymore (or at least I didn't get it to work).

Not completely. It still works fine to check if a slot was passed or not, which is enough in most cases. It doesn't work only to check if a passed slot has some content or not.

I understand using `isSlotPopulated` in our components like `<NcButton>` where we actually get text content from the slot. When it is not about text content, but with things like buttons/icons that are not generated, I'd prefer to force users to not pass empty slots.

```vue
<!-- simple check doesn't work because slots are passed but have no real content -->
<Comp>
  <template #foo>
    <!-- nothing -->
    <div v-if="false">Not rendered</div>
    {{ mayBeEmptyString }}
  </template>
  <template #bar></template>
</Comp>

<!-- But we should not use v-if on content when slot is optional -->
<Comp>
  <template #icon>
    <Icon v-if="withIcon" />
  </template>
</Comp>

<!-- Instead, we can use v-if to define is a slot is to be passed, and it works -->
<Comp>
  <template v-if="withIcon" #icon>
    <Icon />
  </template>
</Comp>
```

And I don't consider this a breaking change, because it's a Vue's change, not `NcInputField`'s

Reply:
> @ShGKme See #4516.

Not sure what to see there...

"Both `!!this.$slots.default` and `!!this.$slots.default?.()` will basically always be true" - only for passed slot that we want to check for emptiness. Not for checking if a slot is passed:

![check-slots](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/f00c035c-4bb1-4c9b-b99d-7b3c1e07f3d0)

https://replit.com/@ShGKme/Vue3-Check-slots

Reply:
```suggestion
			return !!this.$slots.default
```

Reply:
This does not work with the way we use `NcInputField` in `NcTextField` and `NcPasswordField`:

| `return !!this.$slots.default` (always true): | `return isSlotPopulated(this.$slots.default?.())` (works correctly):
|-|-|
| ![grafik](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/72665e92-24c9-4c61-8837-036bf93e0331) | ![grafik](https://github.com/nextcloud-libraries/nextcloud-vue/assets/2496460/a0ddb9b8-511a-43ff-97ab-9d9480a582eb) |

I am open to any other working solution, but simply using `return !!this.$slots.default` here does not seem to do the trick. Or I am missing something.

Reply:
Hmm, indeed. Let me check again.